src/bin/tikv-fail.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(plugin)]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+/// Inject failures to `TikV`.
+///
+/// TODO: Integrate into tikv-ctl.
+///
+/// # Examples
+///
+/// ```sh
+/// ./tikv-fail -a 127.0.0.1:9090 inject\
+//      -f tikv::raftstore::store::store::raft_between_save\
+///     -a "panic(fail_point_raft_between_write)"
+/// ```
+///
+
+extern crate tikv;
+extern crate clap;
+extern crate grpcio as grpc;
+extern crate protobuf;
+extern crate kvproto;
+
+use std::fs;
+use std::io::Read;
+use std::str;
+use std::time::Duration;
+use std::sync::Arc;
+
+use clap::{App, Arg, ArgMatches, SubCommand};
+use grpc::{CallOption, ChannelBuilder, EnvBuilder};
+use kvproto::debugpb;
+use kvproto::debugpb_grpc::DebugClient;
+
+fn main() {
+    let app = App::new("TiKV fail point")
+        .author("PingCAP")
+        .about("A tool for injecting failures to TiKV and recovery")
+        .arg(
+            Arg::with_name("addr")
+                .short("a")
+                .takes_value(true)
+                .help("set tikv ip:port"),
+        )
+        .subcommand(
+            SubCommand::with_name("inject")
+                .about("Inject failures")
+                .arg(
+                    Arg::with_name("failpoint")
```

Comment:
I think just support list should be ok.

Reply:
I think `tikv-fail inject point1 actions point2 actions` is simpler. And then `-f` can be use as short flag for `file`.

Reply:
`tikv-fail inject p1 a1 p2 a2` is hard to achive with clap, see demo[1]. Also I think batch injecting with a file is more simpler. If you want `f` for `file`, I can set `n` for "failpoint", "n" means "name".

[1] https://play.rust-lang.org/?gist=27f38b67c051359a6e28c3dc6e66c4dd&version=stable

Reply:
I think that's because an additional arg is missing. For example, you can add following to the subcommand:

```
.arg(
    Arg::with_name("args")
        .takes_value(true)
        .multiple(true),
)
```

Reply:
`inject` is not an `Arg`, but an `App`.

See more: 
 - https://docs.rs/clap/2.26.2/clap/struct.SubCommand.html
 - https://docs.rs/clap/2.26.2/clap/struct.App.html
 - https://docs.rs/clap/2.26.2/clap/struct.Arg.html

Reply:
> `inject` is not an `Arg`

I didn't say that. What I said was `.arg(...)` could be added to subcommand `inject` to accept positional args.

> ..., but an App.

I don't get it. So `SubCommand::with_name("inject")` at L58 create an `App` instead of a `SubCommand`?

Reply:
Interesting, it turns out that `SubCommand::with_name` will return an `App` instead of a `SubCommand`. Very weird API design. But the approach I proposed should still work.